lib/upload_creator.rb

```diff
@@ -427,4 +427,25 @@ def add_metadata!
     @upload.secure = UploadSecurity.new(@upload, @opts).should_be_secure?
   end
 
+  private
+
+  def animated?(file)
+    # FastImage will return nil if it cannot determine if animated
+    is_animated = FastImage.animated?(file)
+    return is_animated if is_animated != nil
+
+    # Only GIFs, WEBPs and a few other unsupported image types can be animated
+    type = @image_info.type.to_s
+    return false if type != "gif" && type != "webp"
+
+    OptimizedImage.ensure_safe_paths!(file.path)
+
+    @frame_count ||= begin
```

Comment:
Maybe instead of memoizing just the frame count we could save the resulting boolean? (in `@animated`)